cmds/webboot/webboot.go

```diff
@@ -128,26 +149,14 @@ func main() {
 		verbose = log.Printf
 	}
 
-	cachedDir := *dir
-
-	if cachedDir == "" {
-		mp, err := getCachedDirectory()
-		if err != nil {
-			log.Fatalf("Fail to find the USB stick: %+v", err)
```

Comment:
Looks like you've made some good progress towards figuring out the back option here - I hate to bring this up once you've already done this work, but I had imagined that this could be simpler for the user if you just, instead of having a log.Fatalf when you can't find a cache, just create entries without the "Use Cached ISO" option. So basically only like a 2-3 line change.

Then later once we have 1) the ability to cache a downloaded ISO and 2) a back option, we can revisit this and make it so that people can download an ISO, cache it, and then come back to the main menu and see this "Use Cached ISO" option.

Does that make sense? I think it will make things simpler.

Reply:
That's a good idea, thank you!

Reply:
Sorry I am i little bit not understand, because I have replaced the log.Fatal by a menu with only download option? Is that what you want?

Reply:
ok I think I get you, but the process I imaged is , after the back option is ready, there will always be two option in the main menu, and if the cache dir is exist, the next menu contains the folders and files under the cache dir plus a back option,  and if the cache dir in not exist, the next menu only contains a back option.